app/helpers/route_overrides_helper.rb

```diff
@@ -0,0 +1,40 @@
+module RouteOverridesHelper
+  class << self
+    private
+
+    def mapping_for(from, to)
+      {
+        from.to_s.singularize => to.to_s.singularize,
+        from.to_s.pluralize => to.to_s.pluralize
+      }
+    end
+
+    def map_route_helpers_with(mapping)
+      ['_path', '_url'].each do |suffix|
+        ['', 'new_', 'edit_'].each do |prefix|
+          mapping.each do |from, to|
+            define_method(prefix + from + suffix) do |*forwarded_args|
+              send(prefix + to + suffix, *forwarded_args)
+            end
+          end
+        end
+      end
+    end
+
+    # Override route helper methods e.g. to remove the namespacing in
+    # the model class.
+    #
+    # @param from [Symbol, String] The route helper to be overridden. This helper could be generated
+    #        by a form helper link_to but is not actually created from the route setup.
+    # @param to [Symbol, String] The correct route to be used which is created by the route setup.
```

Comment:
Yes, I removed it because it's complicated and not used in the Feiyue repo. Maybe I'm not aware of the necessary use cases but with the standard RESTful route, I think there's no harm generating both forms.